python/python/lance/ray/distribute_task.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+from typing import Any, Callable, Dict, List
+
+from ray.data._internal.datasource.lance_datasource import LanceDatasource
+
+import lance
+
+
+class TaskCommitMessage:
+    def __init__(self, task_id: str, partition: Dict[Any, Any]):
+        self.task_id = task_id
+        self.partition = partition
+
+    def __getitem__(self, key):
+        if key == "task_id":
+            return self.task_id
+        elif key == "partition":
+            return self.partition
+        else:
+            raise KeyError(f"Key {key} not found")
+
+
+class TaskInput:
+    def __init__(self, task_id: str, fn: Callable, partition: Dict[Any, Any]):
+        self.task_id = task_id
+        self.fn = fn
+        self.partition = partition
+
+
+class CustomTask:
```

Comment:
`CustomTask` may be hard to understand. How about use `RayBaseTask` for these functions?